pkg/tasks/prometheus.go

```diff
@@ -36,7 +38,21 @@ func NewPrometheusTask(client *client.Client, factory *manifests.Factory, config
 	}
 }
 
-func (t *PrometheusTask) Run(ctx context.Context) error {
+func (t *PrometheusTask) Run(ctx context.Context) *StateError {
+	createErr := degradedError(t.create(ctx))
+	validationErr := t.validate(ctx)
+
+	return client.MergeStateErrors(createErr, validationErr)
+}
+
+func (t *PrometheusTask) RunX(ctx context.Context) *StateError {
+	createErr := degradedError(t.create(ctx))
+	validationErr := t.validate(ctx)
+
+	return mergeErrors(createErr, validationErr)
+}
```

Comment:
yes. I thought the refactor or the Tasks's validation and creation as 2 separate tasks can be another refactor PR instead of piggy backing on this bug. WDYT?